checker/src/main/java/org/checkerframework/checker/nullness/NullnessVisitor.java

```diff
@@ -259,6 +276,33 @@ protected void commonAssignmentCheck(
         super.commonAssignmentCheck(varType, valueType, valueTree, errorKey, extraArgs);
     }
 
+    /**
+     * Report initialization errors for the specified tree if the init checekr is active.
+     *
+     * @param tree the tree to report init errors for.
+     * @see InitializationChecker
+     * @see InitializationAnnotatedTypeFactory#reportInitializionErrors
+     */
+    protected void reportInitializationErrors(Tree tree) {
+        if (initFactory == null) {
+            // init checker is deactivated.
+            return;
+        }
+
+        initFactory.reportInitializionErrors(
+                tree,
+                atypeFactory,
+                classTree,
+                AnnotationMirrorSet.singleton(atypeFactory.NONNULL),
+                var -> !atypeFactory.getAnnotatedType(var).getKind().isPrimitive());
+    }
+
+    @Override
+    public Void scan(@Nullable Tree tree, Void p) {
+        reportInitializationErrors(tree);
```

Comment:
Can all trees have an initialization error? Can this be limited to the trees that can have errors, to avoid at least an hashmap lookup on every tree.